apollo-router/src/plugins/telemetry/config_new/attributes.rs

```diff
@@ -0,0 +1,702 @@
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<AttributeValue>,
+    },
+    Baggage {
+        /// The name of the baggage item.
+        baggage: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    Env {
+        /// The name of the environment variable
+        env: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, JsonSchema, Clone, Debug)]
+#[serde(deny_unknown_fields, rename_all = "snake_case", untagged)]
+pub(crate) enum SubgraphCustomAttribute {
+    SubgraphOperationName {
+        /// The operation name from the subgraph query.
+        subgraph_operation_name: OperationName,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    SubgraphOperationKind {
+        /// The kind of the subgraph operation (query|mutation|subscription).
+        subgraph_operation_kind: OperationKind,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    SubgraphQueryVariable {
+        /// The name of a subgraph query variable.
+        subgraph_query_variable: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    SubgraphResponseBody {
+        /// The subgraph response body json path.
+        subgraph_response_body: String,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    SubgraphRequestHeader {
+        /// The name of the subgraph request header.
+        subgraph_request_header: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    SubgraphResponseHeader {
+        /// The name of the subgraph response header.
+        subgraph_response_header: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+
+    SupergraphOperationName {
+        /// The supergraph query operation name.
+        supergraph_operation_name: OperationName,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    SupergraphOperationKind {
+        /// The supergraph query operation kind (query|mutation|subscription).
+        supergraph_operation_kind: OperationKind,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
```

Comment:
same than previously, I think default here is useless

Reply:
[a7d501c](https://github.com/apollographql/router/pull/4061/commits/a7d501cdb0ffad517b7420e328d01b8dcfd8f0b7)